packages/openapi-v3/src/controller-spec.ts

```diff
@@ -235,12 +237,18 @@ function resolveControllerSpec(constructor: Function): ControllerSpec {
  * @param spec Controller spec
  * @param schema Schema object
  */
-function resolveTSType(
+function processSchemaExtensions(
   spec: ControllerSpec,
-  schema?: SchemaObject | ReferenceObject,
+  schema?: SchemaObject | (ReferenceObject & ISpecificationExtension),
 ) {
-  debug('  evaluating schema: %j', schema);
-  if (!schema || isReferenceObject(schema)) return;
+  debug('  processing extensions in schema: %j', schema);
+  if (!schema) return;
+
+  assignRelatedSchemas(spec, schema.definitions);
+  delete schema.definitions;
+
+  if (isReferenceObject(schema)) return;
```

Comment:
Is there a particular reason we split this logic from the early return on line 245?

Reply:
Say you have:
```ts
schema: {
  $ref: #/definitions/Todo,
  definitions: {/* ... */}
}
```
If you keep it the same as before then `!schema || isReferenceObject(schema)` will be a true statement since [`isReferenceObject(schema)`](https://github.com/strongloop/loopback-next/blob/master/packages/openapi-v3-types/src/type-guards.ts#L19-L21) is true and it'll keep the schema as is.

But with the modified check, now it goes through:
```ts
assignRelatedSchemas(spec, schema.definitions);
delete schema.definitions;
```
and assigns related schemas and removes `definitions: {/* ... */}` before returning.

Reply:
Thanks for clarifying @nabdelgadir :-). Sounds good to me.